cluster-cm/src/test/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerDeployClientConfigProviderTest.java

```diff
@@ -0,0 +1,329 @@
+            TimeoutException, IOException {
+        // GIVEN
+        Request request = createDefaultRequest();
+        doReturn(executorService).when(underTest).createExecutor();
+        given(executorService.submit(any(Callable.class))).willReturn(future);
+        given(future.get(120, TimeUnit.SECONDS)).willThrow(new TimeoutException());
+        given(clustersResourceApi.listActiveCommandsWithHttpInfo(anyString(), isNull()))
+                .willReturn(createApiCommandListResponse(404));
+        given(clustersResourceApi.getApiClient()).willReturn(apiClient);
+        given(apiClient.getHttpClient()).willReturn(okHttpClient);
+        given(apiClient.buildRequest(any(), any(), any(), isNull(),
+                any(), any(), any(), isNull())).willReturn(request);
+        given(okHttpClient.newCall(any())).willReturn(requestCall);
+        given(requestCall.execute()).willReturn(
+                createResponse(404, createResponseDefaultString(), request));
+        // WHEN
+        CloudbreakException exception = assertThrows(CloudbreakException.class,
+                () -> underTest.deployClientConfigAndGetCommandId(clustersResourceApi, stack));
+        // THEN
+        assertTrue(exception.getMessage().contains("returned with status code: 404"));
+        verify(executorService, times(1)).shutdown();
+    }
+
+    @Test
+    public void testGetDeployClientConfigCommandIdByCommandTableInvalidResponse()
+            throws ApiException, InterruptedException, ExecutionException,
+            TimeoutException, IOException {
+        // GIVEN
+        Request request = createDefaultRequest();
+        doReturn(executorService).when(underTest).createExecutor();
+        given(executorService.submit(any(Callable.class))).willReturn(future);
+        given(future.get(120, TimeUnit.SECONDS)).willThrow(new TimeoutException());
+        given(clustersResourceApi.listActiveCommandsWithHttpInfo(anyString(), isNull()))
+                .willReturn(createApiCommandListResponse(404));
+        given(clustersResourceApi.getApiClient()).willReturn(apiClient);
+        given(apiClient.getHttpClient()).willReturn(okHttpClient);
+        given(apiClient.buildRequest(any(), any(), any(), isNull(),
+                any(), any(), any(), isNull())).willReturn(request);
+        given(okHttpClient.newCall(any())).willReturn(requestCall);
+        given(requestCall.execute()).willReturn(
+                createResponse(200, "invalid", request));
+        // WHEN
+        CloudbreakException exception = assertThrows(CloudbreakException.class,
+                () -> underTest.deployClientConfigAndGetCommandId(clustersResourceApi, stack));
+        // THEN
+        assertTrue(exception.getMessage().contains("error during processing"));
+        verify(executorService, times(1)).shutdown();
+    }
+
+    private ApiCommand createCommand() {
+        ApiCommand command = new ApiCommand();
+        command.setId(new BigDecimal(1L));
```

Comment:
use a more random value for easy differentiation

Reply:
ok, make sense, changed this value to 7, also changed the responses from listActiveCommands to 4,5,6 (and keep the responses for /cmf/commands/commandTable to 1,2,3) - this way it's more clear which operation returned the value